### ISSUE TITLE
Focus is lost when hitting tag when using transition

### ISSUE THREAD

**Issue by tatianakharitonovich** [tailwindlabs/headlessui#3032]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

 v1.7.19

**What browser are you using?**

Chrome

**Reproduction URL**

https://drive.google.com/file/d/1aDdRnYKZeRmOC4X5uzvlpos3gBPAr4N1/view?usp=sharing

**Describe your issue**

I'm using `Menu (Dropdown)`. 
`MenuItems` wrapped in 
```
<transition
       enter-active-class="transition duration-100 ease-out"
       enter-from-class="transform scale-95 opacity-0"
       enter-to-class="transform scale-100 opacity-100"
       leave-active-class="transition duration-75 ease-out"
       leave-from-class="transform scale-100 opacity-100"
       leave-to-class="transform scale-95 opacity-0"
>
```
When we click on _tag_ the focus is on the `MenuButton`, then when we use the _arrow_ we can navigate inside the `MenuItems`, but then when we click on _tag_ the focus is lost and only after we click on _tag_ again the focus is on the next `Menu (Dropdown)`. Pressing _shift +tag_ does not lose focus. Also, focus loss does not occur in the absence of a transition wrapper.

It is necessary to prevent loss of focus when pressing _tag_.

**Comment by RobinMalfait** [comment]
Hey, thank you for this bug report! 🙏

Unfortunately, I have to close this as a reproduction was not provided. Trying to create a reproduction ourselves from the information in an issue drastically increases the amount of time it takes for us to provide a solution and if we have do that for every issue it’s impossible for us to keep up.

If this is still an issue for you, please open a new issue but include a proper reproduction GitHub repository or CodeSandbox where you already have everything all set up so we can clone it and immediately see the issue without configuring everything by hand.

Sorry, not trying to be dismissive and definitely want to fix the issue if there is one, but we have to be strict about this one or there's no way we could keep up with the number of questions we get and the quality of the library would suffer as a result. Hope that's understandable!
